frontend/public/kubevirt/components/utils/lazyRenderer.jsx

```diff
@@ -0,0 +1,41 @@
+import React from 'react';
+import * as _ from 'lodash-es';
+import { inject } from '../../../components/utils';
+
+const DEFAULT_THROTTLE_DELAY = 500; // in ms
+
+/**
+ * Avoids eager re-rendering for frequent redux store changes.
+ *
+ * Assumption: re-render requests are frequent, so skipping a few of them
+ * has no impact on rendered information from user's perspective.
+ *
+ * If that assumption is not met, the use of this component is not recommended.
+ * Or the component needs to be modified to ensure the render() phase is reliably called, ie.
+ * by updating inner state to ensure the shouldComponentUpdate() function is called.
+ */
+export class LazyRenderer extends React.Component {
+  constructor(props) {
+    super(props);
+    this.setRenderEnabled = this._setRenderEnabled.bind(this);
+    this.setRenderEnabled(false);
+
+    this.enableRendering = _.throttle(() => {
+      this.setRenderEnabled(true);
+    }, DEFAULT_THROTTLE_DELAY);
+  }
+
+  _setRenderEnabled(value) {
+    this.renderEnabled = value;
+  }
+
+  shouldComponentUpdate() {
+    this.enableRendering();
+    return this.renderEnabled;
+  }
+
+  render() {
+    this.setRenderEnabled(false);
```

Comment:
couldn't we set this rather in `shouldComponentUpdate`?

Reply:
will do